sirepo/package_data/static/js/ml.js

```diff
@@ -1943,20 +1898,188 @@ SIREPO.viewLogic('partitionView', function(appState, panelState, $scope) {
     ];
 });
 
-SIREPO.viewLogic('dataFileView', function(appState, panelState, $scope) {
+SIREPO.viewLogic('dataFileView', function(appState, panelState, persistentSimulation, requestSender, validationService, $rootScope, $scope) {
+
+    const modelName = $scope.modelName;
+    const self = this;
+
+    function computeColumnInfo() {
+        const dataFile = appState.models.dataFile;
+        if (! dataFile.file) {
+            appState.models.columnReports = [];
+            appState.saveChanges('columnReports');
+            return;
+        }
+        if (dataFile.file === dataFile.oldFile) {
+            return;
+        }
+        dataFile.oldFile = dataFile.file;
+        appState.saveQuietly('dataFile');
+        requestSender.sendStatefulCompute(
+            appState,
+            function(data) {
+                appState.models.columnInfo = data;
+                computeDefaultPartition();
+                appState.models.columnReports = [];
+                appState.saveChanges(['columnInfo', 'columnReports', 'partition']);
+            },
+            {
+                method: 'compute_column_info',
+                dataFile: dataFile,
+            }
+        );
+    }
+
+    function computeDefaultPartition() {
+        const size = appState.models.columnInfo.rowCount;
+        const partition = appState.models.partition;
+        if (! partition.cutoff0 || ! partition.cutoff1
+            || partition.cutoff0 > size
+            || partition.cutoff1 > size) {
+            partition.cutoff0 = parseInt(0.125 * size);
+            partition.cutoff1 = parseInt((1 - 0.125) * size);
+        }
+    }
+
+    function dataFileChanged() {
+        computeColumnInfo();
+        const dataFile = appState.models.dataFile;
+        const partition = appState.models.partition;
+        if (dataFile.appMode === 'regression'
+            && partition.training + partition.testing >= 100) {
+            ['training', 'testing', 'validation'].forEach(function(f) {
+                delete partition[f];
+            });
+            appState.setModelDefaults(partition, 'partition');
+        }
+        else if (dataFile.appMode === 'classification') {
+            if (partition.training + partition.testing < 100) {
```

Comment:
This deserves a comment. Why the fixup?

Reply:
One for @moellep. I just moved these functions bodily from controller to viewLogic.

Reply:
The fix ups are to ensure that training percentage + testing percentage == 100 for classification sims. For regression (training + testing + validation) == 100. If you switch a sim from classification to/from regression, the percentages need to get adjusted or invalidated.

Reply:
Still think it needs a comment, e.g.


```py
            if (validation > 0) {
                partition.validation = parseFloat(validation.toFixed(2));
            }
```

Why is partition.validation not set to zero when validation is 0?